xija/model.py

```diff
@@ -51,6 +53,41 @@ def convert_type_star_star(array, ctype_type):
     return (f4ptr * len(array))(*[row.ctypes.data_as(f4ptr) for row in array])
 
 
+def _get_bad_times(
+    times: np.ndarray,
+    datestart: str,
+    datestop: str,
+    bad_times_in: Optional[List[Tuple[str, str]]] = None,
+) -> Tuple[List[Tuple[str, str]], List[Tuple[int, int]]]:
+    """Return bad_times, bad_times_indices into ``times`` for elements in the
+    ``bad_times_in`` list that overlap with the ``datestart`` to ``datestop``.
+
+    NOTE: bad_times_in is a list of [datestart, datestop] lists. The "time"
+    name is unfortunate since it has string dates, not float CXCsec times.
+
+    :returns: bad_times: List[List[str, str]]], bad_times_indices: List[List[int, int]]
+    """
+    if bad_times_in is None or len(bad_times_in) == 0:
+        return [], []
+
+    bad_times: np.ndarray = np.array(bad_times_in)
+
+    # Get inclusive overlap of bad_times with datestart to datestop
+    ok = (bad_times[:, 1] > datestart) & (bad_times[:, 0] < datestop)
```

Comment:
I have to admit I was a bit surprised to see that this works, but my it does! #TIL

Reply:
This is more obvious if you start with the logical converse for a single interval:
```
no_overlap = time1 < datestart or time0 > datestop
overlap  =  not(time1 < datestart) and not(time0 > datestop)
overlap = time1 >= datestart and time0 <= datestop
```
And writing it out that way points out that I need the equality in there.